HAL: `quicklogic` includes are broken with `C++` support

**Describe the bug**

`eoss3_dev.h` from the quicklogic HAL is missing code to close out the `extern "C"` boilerplate
https://github.com/zephyrproject-rtos/hal_quicklogic/blob/ba7e35f175fcd28199de11f955cb995e1ef20c73/HAL/inc/eoss3_dev.h#L27-L29

This causes build failures when `__cplusplus` is defined.

**To Reproduce**

`west build -p -b quick_feather samples/basic/hash_map -T libraries.hash_map.newlib.cxx_unordered_map.djb2`

```
/home/jordan/code/zephyr/zephyr/lib/hash/hash_map_cxx.cpp:157:2: error: expected '}' at end of input
  157 | }
      |  ^
In file included from /home/jordan/code/zephyr/zephyr/soc/quicklogic/eos_s3/soc.h:11,
                 from /home/jordan/code/zephyr/zephyr/modules/cmsis/./cmsis_core_m.h:24,
                 from /home/jordan/code/zephyr/zephyr/modules/cmsis/./cmsis_core.h:10,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/arch/arm/asm_inline_gcc.h:24,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/arch/arm/asm_inline.h:18,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/arch/arm/arch.h:33,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/arch/cpu.h:19,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/kernel_includes.h:36,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/kernel.h:17,
                 from /home/jordan/code/zephyr/zephyr/include/zephyr/sys/hash_map.h:20,
                 from /home/jordan/code/zephyr/zephyr/lib/hash/hash_map_cxx.cpp:10:
/home/jordan/code/zephyr/modules/hal/quicklogic/HAL/inc/eoss3_dev.h:28:13: note: to match this '{'
   28 |  extern "C" {
      |             ^
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/jordan/code/zephyr/zephyr/build
```

**Impact**
CI failures

**Environment (please complete the following information):**
 - be682e22e166c33a67c1db0755f8ca20c37b6576

--- a/west.yml
+++ b/west.yml
@@ -203,7 +203,7 @@ manifest:
       groups:
         - hal
     - name: hal_quicklogic
-      revision: b3a66fe6d04d87fd1533a5c8de51d0599fcd08d0
+      revision: bad894440fe72c814864798c8e3a76d13edffb6c
       path: modules/hal/quicklogic
       repo-path: hal_quicklogic
       groups:

